airbyte-commons/src/main/java/io/airbyte/commons/util/CompositeIterator.java

```diff
@@ -32,20 +37,24 @@
  *
  * @param <T> type
  */
-public final class CompositeIterator<T> extends AbstractIterator<T> implements AutoCloseableIterator<T> {
+public final class CompositeIterator<T> extends AbstractIterator<T> implements AutoCloseableIterator<T>, AirbyteStreamAware {
```

Comment:
@cgardens It feels like airbyte-commons should go away entirely in this repo now that it has been separated from the platform.  This code is only being used by the connectors anyway now in this repository.  Happy to move this, but it seems like the connector teams should do some cleanup on this repo post repository split.

Reply:
@evantahler Tagging you here for awareness.

I attempted to do what @cgardens suggested by moving the code that now knows about streams out of airbyte-commons and into the base-java module.  However, I stopped when I realized that moving some of the iterators in question would cause other classes to need to move with them, like the `Yamls.java`.  That refactor lead down a rabbit hole to the `airbyte-config-oss` module, which in turn started to cause a circular dependency if classes from that module also needed to move.

I think what this is telling us is that since the repository split, there is some overdue refactoring to clean up the modules inherited from when connectors and platform were in the same repository.  In other words, most of this code should probably move into a "base" module under airbyte-integrations to help clean this up and remove any doubt that these modules are no longer common or shared but only used by the connectors in this repository.